src/freertos_drivers/esp32/Esp32WiFiConfiguration.hxx

```diff
@@ -0,0 +1,139 @@
+    /// the wifi_sleep and hub_mode fields.
+    static constexpr const char *BOOLEAN_MAP =
+        "<relation><property>0</property><value>No</value></relation>"
+        "<relation><property>1</property><value>Yes</value></relation>";
+
+    /// Visible name for the WiFi Power Savings mode.
+    static constexpr const char *WIFI_POWER_SAVE_NAME =
+        "WiFi Power Savings Mode";
+
+    /// Visible description for the WiFi Power Savings mode.
+    static constexpr const char *WIFI_POWER_SAVE_DESC = 
+        "When enabled this allows the ESP32 WiFi radio to use power savings "
+        "mode which puts the radio to sleep except to receive beacon updates "
+        "from the connected SSID. This should generally not need to be "
+        "enabled unless you are powering the ESP32 from a battery.";
+
+    /// Visible name for the Hub Configuration group.
+    static constexpr const char *HUB_NAME = "Hub Configuration";
+
+    /// Visible description for the Hub Configuration group.
+    static constexpr const char *HUB_DESC = 
+        "Configuration settings for an OpenLCB Hub";
+
+    /// Visible name for the hub enable field.
+    static constexpr const char *HUB_ENABLE_NAME = "Enable Hub Mode";
+
+    /// Visible description for the hub enable field.
+    static constexpr const char *HUB_ENABLE_DESC = 
+        "Defines this node as a hub which can accept connections";
+
+    /// Visible name for the hub_listener_port field.
+    static constexpr const char *HUB_LISTENER_PORT_NAME =
+        "Hub Listener Port";
+
+    /// Visible name for the hub_listener_port field.
+    static constexpr const char *HUB_LISTENER_PORT_DESC = 
+        "Defines the TCP/IP listener port this node will use when operating "
+        "as a hub. Most of the time this does not need to be changed.";
+
+    /// Visible name for the link_config group.
+    static constexpr const char *UPLINK_NAME = "Node Uplink Configuration";
+
+    /// Visible name for the link_config group.
+    static constexpr const char *UPLINK_DESC =
+        "Configures how this node will connect to other nodes.";
+};
+
+/// CDI Configuration for an @ref Esp32WiFiManager managed hub.
+CDI_GROUP(HubConfiguration);
+/// Allows the node to become a Grid Connect Hub.
+CDI_GROUP_ENTRY(enable, openlcb::Uint8ConfigEntry,
+    Name(Esp32WiFiConfigurationParams::HUB_ENABLE_NAME),
+    Description(Esp32WiFiConfigurationParams::HUB_ENABLE_DESC),
+    Min(0), Max(1), Default(0),
+    MapValues(Esp32WiFiConfigurationParams::BOOLEAN_MAP));
+/// Specifies the port which should be used by the hub.
+CDI_GROUP_ENTRY(port, openlcb::Uint16ConfigEntry,
+    Name(Esp32WiFiConfigurationParams::HUB_LISTENER_PORT_NAME),
+    Description(Esp32WiFiConfigurationParams::HUB_LISTENER_PORT_DESC),
+    Min(1), Max(65535),
+    Default(openlcb::TcpClientDefaultParams::DEFAULT_PORT))
+/// Specifies the mDNS service name to advertise for the hub.
+CDI_GROUP_ENTRY(service_name, openlcb::StringConfigEntry<48>,
```

Comment:
Consider adding a few bytes (4-6) of reserved space before this entry. Will help to avoid needing to reset settings in case of a later update.

Reply:
I copied the size from TcpAutoAddress / service_name, should we increase this in all places to a uniform size?

Reply:
the point is not the byte size of service name, but to have a few reserved bytes in the parent structure so we can add more options later.

Reply:
ahh, I'll add a hidden block of bytes so we can inject them if needed in the future.